client/app/components/composites/Menu/Menu.js

```diff
@@ -37,7 +37,7 @@ class Menu extends Component {
     const menuLabel = ReactDOM.findDOMNode(this.menuLabel);
 
     this.setState({ // eslint-disable-line react/no-did-mount-set-state, react/no-set-state
-      arrowPosition: menuLabel.offsetWidth / 2, // eslint-disable-line no-magic-numbers
+      arrowPosition: menuLabel.offsetWidth > 40 ? menuLabel.offsetWidth / 2 : 20, // eslint-disable-line no-magic-numbers
```

Comment:
Yeah, the problem is that arrowPosition must be calculated after the component has connection to DOM. I can add some better wording to the magic numbers (which are just setting the arrow to some minimum distance from dropdown box's corner (rounded corner)